pkg/gardenlet/controller/managedseed/add_test.go

```diff
@@ -0,0 +1,163 @@
+	)
+
+	BeforeEach(func() {
+		reconciler = &Reconciler{
+			GardenNamespace: v1beta1constants.GardenNamespace,
+		}
+		fakeClient = fakeclient.NewClientBuilder().WithScheme(kubernetes.GardenScheme).Build()
+
+		shoot = &gardencorev1beta1.Shoot{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+		}
+
+		managedSeedShoot = &seedmanagementv1alpha1.Shoot{
+			Name: name,
+		}
+	})
+
+	Describe("#ManagedSeedFilterPredicate", func() {
+		var (
+			oldManagedSeed, newManagedSeed *seedmanagementv1alpha1.ManagedSeed
+		)
+
+		BeforeEach(func() {
+			pred = reconciler.ManagedSeedFilterPredicate("test-seed")
+
+			oldManagedSeed = &seedmanagementv1alpha1.ManagedSeed{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+				},
+			}
+
+			newManagedSeed = &seedmanagementv1alpha1.ManagedSeed{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      name,
+					Namespace: namespace,
+				},
+			}
+
+			Expect(inject.StopChannelInto(ctx.Done(), pred)).To(BeTrue())
+			Expect(inject.ClientInto(fakeClient, pred)).To(BeTrue())
+		})
+
+		It("should return false when ManagedSeed does not references any shoot", func() {
+			Expect(pred.Create(event.CreateEvent{Object: newManagedSeed})).To(BeFalse())
+			Expect(pred.Update(event.UpdateEvent{ObjectOld: oldManagedSeed, ObjectNew: newManagedSeed})).To(BeFalse())
+			Expect(pred.Delete(event.DeleteEvent{Object: newManagedSeed})).To(BeFalse())
+			Expect(pred.Generic(event.GenericEvent{})).To(BeFalse())
+		})
+
+		It("should return false when shoot referenced by ManagedSeed is not present", func() {
+			oldManagedSeed.Spec.Shoot = managedSeedShoot
+			newManagedSeed.Spec.Shoot = managedSeedShoot
+			Expect(pred.Create(event.CreateEvent{Object: newManagedSeed})).To(BeFalse())
+			Expect(pred.Update(event.UpdateEvent{ObjectOld: oldManagedSeed, ObjectNew: newManagedSeed})).To(BeFalse())
+			Expect(pred.Delete(event.DeleteEvent{Object: newManagedSeed})).To(BeFalse())
+			Expect(pred.Generic(event.GenericEvent{})).To(BeFalse())
+		})
+
+		It("should return false when shoot referenced by ManagedSeed does not references any seed", func() {
+			oldManagedSeed.Spec.Shoot = managedSeedShoot
+			newManagedSeed.Spec.Shoot = managedSeedShoot
+			Expect(fakeClient.Create(ctx, shoot)).To(Succeed())
+			Expect(pred.Create(event.CreateEvent{Object: newManagedSeed})).To(BeFalse())
+			Expect(pred.Update(event.UpdateEvent{ObjectOld: oldManagedSeed, ObjectNew: newManagedSeed})).To(BeFalse())
+			Expect(pred.Delete(event.DeleteEvent{Object: newManagedSeed})).To(BeFalse())
+			Expect(pred.Generic(event.GenericEvent{})).To(BeFalse())
+		})
+
+		It("should return true when shoot referenced by ManagedSeed references a seed which is same as the seed mentioned in gardenlet configuration", func() {
+			oldManagedSeed.Spec.Shoot = managedSeedShoot
+			newManagedSeed.Spec.Shoot = managedSeedShoot
+			shoot.Spec.SeedName = pointer.String("test-seed")
```

Comment:
Maybe define a variable for "test-seed" and reuse it for better readability of the test?